datashader/core.py

```diff
@@ -123,6 +129,12 @@ def __init__(self, plot_width=600, plot_height=600,
         self.x_axis = _axis_lookup[x_axis_type]
         self.y_axis = _axis_lookup[y_axis_type]
 
+        if x_axis_type=='log' and x_range and x_range[0]<=0:
+            raise ValueError("x_range must be >0 for a log x axis")
+        if y_axis_type=='log' and y_range and y_range[0]<=0:
+            raise ValueError("y_range must be >0 for a log y axis")
```

Comment:
It'd be more ideal to attach this validation to the log-axis object. How I'd do this with minimal changes would be:

- Make `LogAxis` and `LinearAxis` subclasses of `Axis`, with the `mapper` and `inverse_mapper` arguments moved to method definitions (with a `staticmethod` decorator).
- Add a `validate` method to both `Axis` subclasses with appropriate validation
- Call the validate methods for both axis in `bypixel`

The benefit of this is that it's more extensible, and the checks get run if the ranges/axis are mutated after initializing `Canvas`.